apps/codeimage/src/components/Presets/PresetSwitcher/PresetSwitcher.tsx

```diff
@@ -34,11 +35,55 @@ import {EmptyPresetFallback} from '../EmptyPresetFallback/EmptyPresetFallback';
 import {PresetPreview} from '../PresetPreview/PresetPreview';
 import {PresetUpdateDialog} from '../PresetUpdateDialog';
 import * as styles from './PresetSwitcher.css';
+import {getAuth0State} from '@codeimage/store/auth/auth0';
 
 type PresetSwitcherProps = {
   onClose: () => void;
 };
+const env = import.meta.env;
+const userLimit = env.VITE_PRESET_LIMIT;
+const guestLimit = env.VITE_PRESET_LIMIT_GUEST;
+function AddNewPreset() {
+  const presetsStore = getPresetsStore();
+  const openDialog = createControlledDialog();
+  const [t] = useI18n<AppLocaleEntries>();
+  const isLogged = () => getAuth0State().loggedIn();
+  const guestLimitReach = () =>
+    !isLogged() && presetsStore.sortedPresets().length >= guestLimit;
+  const userLimitReach = () =>
+    isLogged() && presetsStore.sortedPresets().length >= userLimit;
 
+  const isSomeLimitReach = () => guestLimitReach() || userLimitReach();
+  const labelTooltip = () =>
+    isLogged()
+      ? "You can't add more presets, sorry :("
+      : "You can't add more presets, please login to add more presets";
+
+  return (
+    <Tooltip
+      content={labelTooltip()}
+      disabled={!isSomeLimitReach()}
```

Comment:
Should be the opposite

```suggestion
      disabled={presetLimitReached()}
```